### ISSUE TITLE
Diary, time should be editable – (Improvement for the available features) 

### ISSUE THREAD

**Issue by hacksenwerk** [davidhealey/waistline#687]
It should be possible to set custom times for each diary entry. 
I do not always fill in the entries, while I am eating.

**Comment by davidhealey** [comment]
You can fill it in whenever you like, I usually fill mine in the day before.

**Comment by hacksenwerk** [comment]
Yes I know, but the time that is shown is always the time when you fill it in. 
For example if I fill in my breakfast at the evening, I would liked to change the time in the entry to the time I had that breakfast.

**Comment by rgrregor** [comment]
I also think this enhancement would be really useful. Providing an option for manual timestamping of the diary entries would lead to more granular insight into the time of day a food was consumed, which would then be beneficial when doing correlation with other biomarkers.

**Comment by taladar** [comment]
Just started using the app today.

This would be useful in general but editing would also be useful for entries created before turning on the time feature since those entries all show "Invalid date"

**Comment by RalMay** [comment]
+1 as the workaround (manually changing the devices time) is a PITA and has side effects on appointments and other things.
**Please** add an edit box for the date (in new-entry-dialog as well as in edit-entry-dialog).
